### PR TITLE
add Classification enum

### PR DESCRIPTION
**Pull request by WalterBright**
Just makes it easier to see what the scheme is.

### CODE REVIEW

**Comment by UplinkCoder** [comment]
You might as well put the in the color enum.
`Color.error = Color.red`,

File: src/ddmd/errors.d [hunk 1]
```diff
@@ -19,6 +19,16 @@ import ddmd.root.outbuffer;
 import ddmd.root.rmem;
 import ddmd.console;
 
+/**********************
+ * Color highlighting to classify messages
+ */
+enum Classification
+{
+    error = Color.red,          // for errors
```
Comment (jacob-carlborg) [inline src/ddmd/errors.d]: Please use Ddoc comments.

File: src/ddmd/errors.d [hunk 2]
```diff
@@ -19,6 +19,16 @@ import ddmd.root.outbuffer;
...
+    gagged = Color.magenta,     // for gagged errors
+    warning = Color.yellow,     // for warnings
+    _deprecation = Color.blue,  // for deprecations
```
Comment (ZombineDev) [inline src/ddmd/errors.d]: Nit: the convention (at least in Phobos) is `keyword_`, i.e. this should be `deprecation_`.

**Review by PetarKirov — APPROVED** [review]
LGTM, considering my and Jacob's comments are addressed.

File: src/ddmd/errors.d (hunk 2 above)
Comment (wilzbach) [inline src/ddmd/errors.d]: @ZombineDev: how about adding this to the [D Style guide](http://dlang.org/dstyle.html)? The [Vibe.d](https://vibed.org/style-guide) already specifies this to some extent..

**Comment by ZombineDev** [comment]
@wilzbach it's already there - search for "keywords" ;)

File: src/ddmd/errors.d (hunk 2 above)
Comment (jacob-carlborg) [inline src/ddmd/errors.d]: I think the convention in DMD is the opposite, prefixing with underscore, unfortunately. Although I think that should be changed.

**Comment by WalterBright** [comment]
> You might as well put the in the color enum.

Having usage details creep into the colorization module means it will no longer be encapsulated.

**Comment by WalterBright** [comment]
ping @yebblies

**Review by yebblies — APPROVED** [review]
Why does `_deprecation` start with an `_` but the others don't?

**Comment by jacob-carlborg** [comment]
> Why does _deprecation start with an _ but the others don't?

Confusion between `deprecated` (which is a keyword) and `deprecation`?

**Comment by WalterBright** [comment]
> Why does _deprecation start with an _ but the others don't?

`deprecation` is a keyword and the others aren't. Look in `idgen.d` and there are several other examples of this.

**Comment by jacob-carlborg** [comment]
> `deprecation` is a keyword and the others aren't. Look in `idgen.d` and there are several other examples of this.

@WalterBright `deprecation` is not a keyword [1], but `deprecated` is.

[1] https://github.com/dlang/dmd/blob/master/src/ddmd/access.d#L467

**Comment by WalterBright** [comment]
Oops, yeah, you're right! Will fix.

**Pull request merged**